### PR TITLE
Fix StatusBar color

### PR DESCRIPTION
**Pull request by arianeleonard**
GitHub Issue: #

## PR Type

What kind of change does this PR introduce?

- Bugfix

## Description

Fix StatusBar color
Color change depending on the page because of background color changes making the status bar not visible.

## PR Checklist 
Please check if your PR fulfills the following requirements:

- [ ] Interface members are XML documented
- [ ] Documentation (XML or comments) has been added and/or existing documentation has been updated
- [ ] [Architecture documents](./doc/Architecture.md) have been updated
- [x] Tested on all relevant platforms

## Internal Issue (If applicable):

https://dev.azure.com/nventive/Practice%20committees/_backlogs/backlog/Mobile%20(.Net)/Features/?workitem=252413

### CODE REVIEW

File: src/app/ApplicationTemplate.Shared/CoreStartup.cs [hunk 1]
```diff
@@ -236,5 +246,58 @@ protected override ILogger GetOrCreateLogger(IServiceProvider serviceProvider)
 		{
 			return serviceProvider.GetRequiredService<ILogger<CoreStartup>>();
 		}
+
+		private async Task ObserveCurrentViewModel(CancellationToken ct, IServiceProvider services)
```
[the remark this answers is not in the document]
Comment (arianeleonard) [reply src/app/ApplicationTemplate.Shared/CoreStartup.cs:250]: I'll add a comment as well to explain this change

File: src/app/ApplicationTemplate.Shared/CoreStartup.cs [hunk 2]
```diff
@@ -236,5 +246,58 @@ protected override ILogger GetOrCreateLogger(IServiceProvider serviceProvider)
...
+		{
+			var dispatcher = services.GetRequiredService<CoreDispatcher>();
+
+			services
+				.GetRequiredService<ISectionsNavigator>()
+				.ObserveProcessedState()
+				.Select(async (state) =>
+				{
+					var vmType = state.CurrentState.GetViewModelType();
+					if (Window.Current.Content is FrameworkElement root)
```
Comment (carlh98) [inline src/app/ApplicationTemplate.Shared/CoreStartup.cs:260]: I would combine the 2 ifs

Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Shared/CoreStartup.cs:260]: You should not reference any View from `CoreStartup`. Use `Startup` instead.

File: src/app/ApplicationTemplate.Shared/CoreStartup.cs [hunk 3]
```diff
@@ -58,6 +59,15 @@ protected override async Task StartServices(IServiceProvider services, bool isFi
 
 				services.GetRequiredService<DiagnosticsCountersService>().Start();
 
+//-:cnd:noEmit
+#if __ANDROID__ || __IOS__
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Shared/CoreStartup.cs:63]: Wouldn't it be cleaner if this condition was in the method rather than outside of it?

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

File: src/app/ApplicationTemplate.Shared.Views/Startup.cs [hunk 4]
```diff
@@ -119,5 +124,59 @@ protected override ILogger GetOrCreateLogger(IServiceProvider serviceProvider)
...
+
+		private async Task SetStatusBarColor(CancellationToken ct, IServiceProvider services)
+		{
+//-:cnd:noEmit
+#if __ANDROID__ || __IOS__
+//+:cnd:noEmit
+			var dispatcher = services.GetRequiredService<CoreDispatcher>();
+
+			// These are pages with a different background color, needing a different status bar color
+			Type[] vmExceptions =
```
Comment (carlh98) [inline src/app/ApplicationTemplate.Shared.Views/Startup.cs:136]: Maybe vmAlternateColor?

File: src/app/ApplicationTemplate.UWP/Views/Content/Settings/SettingsPage.xaml.cs [hunk 5]
```diff
@@ -20,9 +20,23 @@ private void OnThemeButtonClicked(object sender, Windows.UI.Xaml.RoutedEventArgs
 					case ElementTheme.Default:
 					case ElementTheme.Light:
 						root.RequestedTheme = ElementTheme.Dark;
+						//-:cnd:noEmit
+#if __ANDROID__ || __IOS__
+						//+:cnd:noEmit
+						Windows.UI.ViewManagement.StatusBar.GetForCurrentView().ForegroundColor = Windows.UI.Colors.Black;
```
Comment (carlh98) [inline src/app/ApplicationTemplate.UWP/Views/Content/Settings/SettingsPage.xaml.cs:26]: No need for the dispatcher here?

File: src/app/ApplicationTemplate.Shared.Views/Startup.cs [hunk 6]
```diff
@@ -119,5 +124,59 @@ protected override ILogger GetOrCreateLogger(IServiceProvider serviceProvider)
 		{
 			return serviceProvider.GetRequiredService<ILogger<Startup>>();
 		}
+
+		private async Task SetStatusBarColor(CancellationToken ct, IServiceProvider services)
```
Comment (jeanplevesque) [inline src/app/ApplicationTemplate.Shared.Views/Startup.cs:128]: You can probably remove the CancellationToken parameter because it's not used

**Review by carlh98 — APPROVED** [review]

**Review by jeanplevesque — APPROVED** [review]

**Pull request merged**